app/src/main/java/com/jerboa/db/AppDB.kt

```diff
@@ -242,6 +290,7 @@ class AppSettingsRepository(
     @WorkerThread
     suspend fun update(appSettings: AppSettings) {
         appSettingsDao.updateAppSettings(appSettings)
+        if (!appSettings.saveSearchHistory) searchHistoryDao.clear()
```

Comment:
IMO this isn't necessary, but its up to you. The saved search history doesn't really need to clear anything, it just needs to visually not show it.

Reply:
I wanted it, to be added because if you don't want your searches be saved it shouldn't still have them saved in the DB

Reply:
I've kept this for now. Lmk if it needs to be removed,